adafruit_emc2101/emc2101_lut.py

```diff
@@ -81,7 +110,8 @@ def __setitem__(self, index, value):
             raise AttributeError("LUT values must be a fan speed from 0-100%")
         else:
             self.lut_values[index] = value
-        self._update_lut()
+        if self._defer_update == 0:
```

Comment:
This bit isn't about nesting. Even without nesting control, _defer_update is a flag that says 'are we supposed to be updating the lut now, or not'. If now, then do it, else we assume it's because we're in a with block and will do the lut update in the exit function.

The reason for this is that it was taking noticeable time to initialise the devices on my board, and I noticed this was because I set the lut up with about 5 standard values and each one caused _update_lut to be called, which rewrote the registers... quite a lot of activity to repeat 5 times!

Reply:
`__exit__` will run whenever leaving the context manager, so this bit of code seems to keep track of how nested it is.  With nesting removed, most of the logic here can be removed, but the call to `self._update_lut()` should definitely remain.  Basically, my thought is that `__enter__` should be a no-op, and `__exit__` should just be the call to `self._update_lut()`.

Reply:
But you still have to do something within the existing __setitem__ call, which I want to support without 'with' enclosure, which means setitem needs to know if it's been called alone (and so call _update_lut) or as a group (and not call). That at least needs a flag variable, which is essentially what I have.